presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -784,8 +817,11 @@ public Converter getConverter(int fieldIndex)
         @Override
         public void start()
         {
-            out.reset();
-            this.currentBuilder = createBlockBuilder();
+            List<Type> typeParameters = type.getTypeParameters();
+            checkArgument(typeParameters.size() == 1, "list must have exactly 1 type parameter");
+            Type elementType = typeParameters.get(0);
+
+            this.currentBuilder = elementType.createBlockBuilder(new BlockBuilderStatus(), 100);
```

Comment:
It could. But entry writer was started as an optimization for rows. And we discussed the possibility and benefit of also using it for arrays. It should surely be added in the future. I can also do it in this PR if you prefer so.